resolver-dns/src/main/java/io/netty/resolver/dns/DnsNameResolver.java

```diff
@@ -535,18 +553,29 @@ private InetSocketAddress nextNameServerAddress() {
             InetSocketAddress nameServerAddr, DnsQuestion question,
             Promise<AddressedEnvelope<? extends DnsResponse, InetSocketAddress>> promise) {
 
-        return query0(checkNotNull(nameServerAddr, "nameServerAddr"),
-                      checkNotNull(question, "question"),
-                      checkNotNull(promise, "promise"));
+        return query0(nameServerAddr, question, Collections.<DnsRecord>emptyList(), promise);
+    }
+
+    /**
+     * Sends a DNS query with the specified question with additional records using the specified name server list.
+     */
+    public Future<AddressedEnvelope<DnsResponse, InetSocketAddress>> query(
+            InetSocketAddress nameServerAddr, DnsQuestion question,
+            List<DnsRecord> additional,
+            Promise<AddressedEnvelope<? extends DnsResponse, InetSocketAddress>> promise) {
+
+        return query0(nameServerAddr, question, additional, promise);
     }
 
     private Future<AddressedEnvelope<DnsResponse, InetSocketAddress>> query0(
             InetSocketAddress nameServerAddr, DnsQuestion question,
+            List<DnsRecord> additional,
```

Comment:
I have no problem with replacing `List<DnsRecord>` with `Iterable<DnsRecord>` but @slandelle and other guys prefer to usage of clean for-loop without iterator (see comments ten days ago). Question is still same: Which kind of collection is preferred in Netty API?

I prefer immutable collections personally, but standard Java don't provide any usable...